tests/components/blue_current/__init__.py

```diff
@@ -1,42 +1,121 @@
 """Tests for the Blue Current integration."""
 from __future__ import annotations
 
-from unittest.mock import patch
+from asyncio import Event, sleep
+from functools import partial
+from unittest.mock import MagicMock, patch
 
 from bluecurrent_api import Client
 
-from homeassistant.components.blue_current import DOMAIN, Connector
-from homeassistant.config_entries import ConfigEntry
+from homeassistant.components.blue_current import DOMAIN
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.dispatcher import async_dispatcher_send
 
 from tests.common import MockConfigEntry
 
+DEFAULT_CHARGE_POINT = {
+    "evse_id": "101",
+    "model_type": "",
+    "name": "",
+}
+
+
+async def connect(client, token: str):
+    """Set the connected event."""
+    if client.connect_exception:
+        # Raise exception once.
+        temp = client.connect_exception
+        client.connect_exception = None
+        raise temp
+
+    client.connected.set()
+    client.connected.clear()
+
+
+async def start_loop(client, receiver):
+    """Set the receiver."""
+    client.receiver = receiver
+    client.loop_start_task.set()
+
+    while True:
```

Comment:
I'd use an `asyncio.Future` to wait on instead. We can set the result of the future from outside when we want the loop to finish or set an exception on the future if we want the loop to raise an exception.